exporter/metric/metric.go

```diff
@@ -374,12 +370,30 @@ func recordToTypedValueAndTimestamp(r *export.Record) (*monitoringpb.TypedValue,
 			StartTime: &googlepb.Timestamp{
 				Seconds: now,
 			},
-		}
-		return &monitoringpb.TypedValue{
-			Value: &monitoringpb.TypedValue_Int64Value{
-				Int64Value: value,
+			EndTime: &googlepb.Timestamp{
+				Seconds: now,
 			},
-		}, t, nil
+		}
+		switch kind {
```

Comment:
You have a repetitive code `return ..., t, nil` below.
If you extract this switch-case block into a separate function `countToTypedValue(kind, value)`, the code will be much cleaner:
```go
typedValue := countToTypedValue(kind, value)
return typedValue, t, nil
```

Reply:
You can also do the same for LastValue and Sum below.
Actually would also be nice to extract every if condition there:
```
if count, ok := agg.(aggregator.Count); ok {
  return countToTypedValueAndTimestamp(count, kind)
} else if lv, ok := agg.(aggregator.LastValue); ok {
  return lastValueToTypedValueAndTimestamp(lv, kind)
} else ...
```

Reply:
Thank you, Nail. I extracted all cases to functions as suggested. Actually it looks much cleaner.